pennylane/compiler/compiler.py

```diff
@@ -43,6 +48,12 @@ def _refresh_compilers():
         module_name = entry.module_name.split(".")[0]
         AvailableCompilers.names_entrypoints[module_name][entry.name] = entry
 
+    # Check the definition of all expected entry points in compiler packages
+    for _, eps_dict in AvailableCompilers.names_entrypoints.items():
+        ep_interface = AvailableCompilers.entrypoints_interface
+        if any(ep not in eps_dict.keys() for ep in ep_interface):
+            raise KeyError(f"expected {ep_interface}, but recieved {eps_dict}")
```

Comment:
What does this code block do @maliasadi?

Reply:
It will check whether available compilers follow the `entry_point` interface by validating that all 3 entry points (`qjit`, `context`, and `ops`) are defined.

Reply:
Thanks @maliasadi! I recommend updating the comment to match what you replied to me, as it is more informative :)

Reply:
Sure thing! :100: